packages/hash/frontend/src/pages/[account-slug]/entities/new.page.tsx

```diff
@@ -0,0 +1,68 @@
+import { useRouter } from "next/router";
+import { useState } from "react";
+import { useEffectOnceWhen } from "rooks";
+import { useLoggedInUser } from "../../../components/hooks/useAuthenticatedUser";
+import { PageErrorState } from "../../../components/page-error-state";
+import { getPlainLayout, NextPageWithLayout } from "../../../shared/layout";
+import { mustBeVersionedUri } from "../types/entity-type/util";
+import { EntityPageLoadingState } from "./[entity-uuid].page/entity-page-loading-state";
+import { NewEntityPage } from "./[entity-uuid].page/new-entity-page";
+import { useCreateNewEntityAndRedirect } from "./[entity-uuid].page/shared/use-create-new-entity-and-redirect";
+
+type PageState = "error" | "loading" | "ready";
+
+const Page: NextPageWithLayout = () => {
+  const router = useRouter();
+  const { authenticatedUser } = useLoggedInUser();
+  const createNewEntityAndRedirect = useCreateNewEntityAndRedirect();
+  const [pageState, setPageState] = useState<PageState>("loading");
```

Comment:
The `"error"` version of this is entirely derived state – so you can just derive this at render time. This can then be a `isReady` flag, which should be `true` unless `entity-type-id` is in the query param, and then it can be set to true once the `useLoggedInUser` callback I've suggested below (which replaces `useEffectOnceWhen`) is called. I think that would make this all much clearer.

Reply:
Makes sense!